src/components/tree/Tree.tsx

```diff
@@ -1,79 +1,141 @@
       structure, these should be the first authors to discover the
       applicability of your topic of interest
     `,
   },
   leaf: {
-    title: "Leaves",
+    title: 'Leaves',
     info: `
       Here you should find recent articles and reviews that should
       condense very well your topics.
     `,
   },
 };
 
+const DEFAULT_SHOW = {
+  root: true,
+  trunk: true,
+  leaf: true,
+};
+
 const Tree: FC<{}> = () => {
   const data: { [section: string]: Article[] } = DATA;
+  const [star, setStar] = useState<{ [label: string]: boolean }>({});
+  const [show, setShow] = useState<{ [section: string]: boolean }>(
+    DEFAULT_SHOW
+  );
+
+  const toggleStar = useCallback((label: string) => {
+    setStar((current) => ({ ...current, [label]: !current[label] }));
+  }, []);
+
+  const toggleShow = useCallback((label: string) => {
+    setShow((curr) => {
+      if (
+        Object.values(curr).reduce(
+          (acc, selected) => acc + (selected ? 1 : 0),
+          0
+        ) === 1 &&
+        curr[label]
+      ) {
+        return {
+          root: true,
+          trunk: true,
+          leaf: true,
+        };
+      }
+      return {
+        root: false,
+        trunk: false,
+        leaf: false,
+        [label]: true,
+      };
+    });
+  }, []);
 
   return (
     <Fragment>
-      <div className="tree-menu">
+      <div className='tree-menu'>
         {Object.entries(data).map(([sectionName, articles]) => (
-          <button className={`btn btn-${sectionName} ${sectionName}`}>
-            <strong>{(INFO[sectionName] || { title: "" }).title}</strong>
+          <button
+            className={`btn btn-${sectionName} ${sectionName} ${
+              show[sectionName] ? 'active' : 'inactive'
+            }`}
+            title='Show only trunk'
+            onClick={() => toggleShow(sectionName)}
+            key={`menu-${sectionName}`}>
+            <strong>{(INFO[sectionName] || { title: '' }).title}</strong>
             <small>{articles.length} articles</small>
           </button>
         ))}
       </div>
 
-      {Object.entries(data).map(([sectionName, articles]) => (
-        <div className={`tree-segment ${sectionName}`}>
-          <div className="info">
-            <h2>{(INFO[sectionName] || { title: "" }).title}</h2>
-            <p>
-              {(INFO[sectionName] || { info: "" }).info}
-              Here you should find seminal articles from the original articles
-              of your topic of interest.
-            </p>
-            <p>
-              <strong>Keywords:</strong> keyword, keyword, keyword
-            </p>
-          </div>
-          <div className="articles">
-            {articles.map((article) => (
-              <div className="article">
-                <Reference key={article.label} {...article} />
-                <CopyImage />
-                <StarImgage />
+      {Object.entries(data).map(
+        ([sectionName, articles]) =>
+          !!show[sectionName] && (
```

Comment:
In case of considering the proposal:
```suggestion
          (!show || sectionName === show) && (
```